### ISSUE TITLE
select event and show the time of the event not the minutes

### ISSUE THREAD

**Issue by heberalmeida** [antoniandre/vue-cal-v4#300]
select event and show the time of the event not the minutes
<img width="787" alt="Captura de Tela 2020-06-16 às 12 39 39" src="https://user-images.githubusercontent.com/30800055/84802429-97048100-afce-11ea-93ec-537c5784347d.png">
select event @cell-click="teste" -> console.log('date')
Not - Tue Jun 16 2020 08:28:00 GMT-0400 (Horário Padrão do Amazonas)
Not - Tue Jun 16 2020 08:14:00 GMT-0400 (Horário Padrão do Amazonas)
true - Tue Jun 16 2020 08:00:00 GMT-0400 (Horário Padrão do Amazonas)
true - Tue Jun 16 2020 08:40:00 GMT-0400 (Horário Padrão do Amazonas)
true - Tue Jun 16 2020 09:20:00 GMT-0400 (Horário Padrão do Amazonas)
hors time event

**Comment by giulianimartini** [comment]
Hey @heberalmeida,

Did you look on the [snap to time](https://antoniandre.github.io/vue-cal/#ex--create-events) property? You can set the snap to time to 40.

If it is not a solution in your case, you always can get the minutes and then check if it's more than the grid time. E.g: if the click is at 08:28, the minute 28 is less than minute 40, than you can set it to minute 00. If the click is at 08:14, the same occurs. But if the click is at 09:34, set the minute to be 20.

I did this manual checking in my code, because I coded before the snap to time property arises. Now you can try to use it or do it manually.

Hope it helps.

**Comment by heberalmeida** [comment]
Hey @giulianimartini,

I want to use the @ cell-click = "logEvents ('cell-click', $ event) method" but when I clicked I didn't want to catch the chopped hours, but what I defined in time-step in this case was time-step = "40 "
time-from
and time-to was 8 at 10, I wanted the return like 8:00 when it was clicked inside it and not 8:12 I wanted it to stay round

**Comment by giulianimartini** [comment]
Hey @heberalmeida,

that is exactly what I said about doing it manually.

You can get the time of the click and then round it.

If it's between 8:00 and 8:40 (e.g: 8:12, discard the 12 minutes and set it to 00).

For example:

`@cell-click="method($event)"`

Then in your `method(event)` you can do something like:

```
let horaSelecionada = event.date.formatTime().substring(0,2); //get the hour part of the time
let minutoSelecionado = event.date.formatTime().substring(3,5); //get the minutes part of the time

if (minutoSelecionado >= 0 && minutoSelecionado < 30) {
    this.horaInicialAjustada = horaSelecionada + ":00:00";
} else if (minutoSelecionado >= 30) {
    this.horaInicialAjustada = horaSelecionada + ':30:00';
}
```

I know this is very manual, but without using `snap-to-time`, I think it's the way to do it.

**Comment by antoniandre** [comment]
Hi @heberalmeida,

If on cell click you want to only keep the hours part, and discard the minutes, this is very easy since you have the date returned at click from `@cell-click="onCellClick"`. 

This will return `15` if you click at `15:48`:

```js
onCellClick({ date }) {
    return date.getHours()
}
```

Remember that the returned date is a native Date object so you can apply any date methods on it.
On top of that, Vue Cal injected Date prototypes will help you format a string very easily if you need: `date.format('HH:00')` will return `15:00` if you click at `15:48`.


___

Thanks for your help @giulianimartini, this is great support. 👍

**Comment by giulianimartini** [comment]
No problem @antoniandre. 

But I think what he wants is to get the minutes of the click like 09:32, to be rounded like 09:20, as `time- step` is set to 40.

And of course, instead of `substring` you can get the hour and minute by the Date methods `getHours` and `getMinutes` (like @antoniandre mentioned) and then you can set the minutes to whatever you want and assign it to the hour again to store.

**Comment by heberalmeida** [comment]
Hi,
that's right, handling is very expensive, it would be good if when you clicked you could get the value q is inside the cell, for example
08:00 --- 08:45
08-45 --- 09:30
click was 08:15 - I would like to pick up at 8:00 am because it is within the range, another case, the click occurred at 9:00 am, the range is 08:45

**Comment by antoniandre** [comment]
Sorry @heberalmeida, what you want is not clear.
If neither myself nor @giulianimartini did answer your question, then can you rephrase your question?

**Comment by giulianimartini** [comment]
Hi @heberalmeida,

do it like I said, but instead of using `substring()`, use `getHours()` and `getMinutes()`. It's simpler.

Do the check by conditional `if` for each rounded minute and you will have what you want.

**Comment by heberalmeida** [comment]
![Captura de Tela 2020-06-17 às 08 58 07](https://user-images.githubusercontent.com/30800055/84901223-26ff0500-b079-11ea-9572-472cdbe9d57a.png)
when click on item 1 the position i click it returns with `@cell-click` method a date this date it comes according to where you click example: 2020-06-17 08:18
I would like it to be returned 2020-06-17 08:18
the if and else resolves when the values ​​are 00 or 30 but in my case the configuration is in `:time-step` = "45" or it can be 20 or 15, etc., I would like the value to be returned as follows: time-step, an example in this case would be item 2 marked in red click on 09:55 return that you would like 09:30, or if it was in the middle of them click on 09:01 return 08:45

**Comment by giulianimartini** [comment]
Is the `time-step` property variable or it's fixed?

Because on the original question you said `time-step = 40` and now 45.

@heberalmeida did you tried `snap-to-time`?

Set the `snap-to time` to the value of `time-step` and you will get the result you want.

I tried this here in my code and I could get the right time: 8:00, 8:45, 9:30, etc.

Do this:

```
<vue-cal style="height: 100%"
    ref="vuecal"
    id="vuecal"
    :time-from="8*60"
    :time-to="10*60"
    :time-step="45"
    :snap-to-time="45"
    active-view="day"
    editable-events
    :drag-to-create-event="true">
</vue-cal>
```

Then test by clicking and dragging to create an event. It will always get the rounded time. Trust me. 👌

**Comment by heberalmeida** [comment]
methods solucioned

`const startTime = moment(event).set({
        hour: this.form.inicio,
        minute: '00'
      })
      const endTime = moment(event).set({
        hour: this.form.fim,
        minute: this.form.duracao_consulta
      })

      let timeStops = []

      while (startTime <= endTime) {
        timeStops.push({
          hours: new moment(startTime).format('HH:mm'),
          hour: new moment(startTime).format('HH'),
          min: new moment(startTime).format('mm'),
          value: Number(
            new moment(startTime)
              .format('HH:mm')
              .split(':')
              .join('')
          )
        })
        startTime.add(this.form.duracao_consulta, 'minutes')
      }`

**Comment by heberalmeida** [comment]
methods solucioned
```
const startTime = moment(event).set({
        hour: this.form.inicio,
        minute: '00'
      })
      const endTime = moment(event).set({
        hour: this.form.fim,
        minute: this.form.duracao_consulta
      })
      let timeStops = []
      while (startTime <= endTime) {
        timeStops.push({
          hours: new moment(startTime).format('HH:mm'),
          hour: new moment(startTime).format('HH'),
          min: new moment(startTime).format('mm'),
          value: Number(
            new moment(startTime)
              .format('HH:mm')
              .split(':')
              .join('')
          )
        })
        startTime.add(this.form.duracao_consulta, 'minutes')
      }
```

**Comment by giulianimartini** [comment]
Nice @heberalmeida. 

You have used `moment`.

If your case is solved, close the issue.

**Comment by heberalmeida** [comment]
![Gravação-de-Tela-2020-06-17-às-20 11 44](https://user-images.githubusercontent.com/30800055/84963305-01a0e400-b0d7-11ea-9333-8cb24343d582.gif)
result final

**Comment by antoniandre** [comment]
Great that you managed to do what you wanted @heberalmeida.
Now that I understand, this might be simpler, more performant and does not require Moment:

```js
onCellClick (date) {
    let dateTimeMinutes = date.getHours() * 60 + date.getMinutes()
    dateTimeMinutes -= (dateTimeMinutes % 45) // If 45 is your time-step.
    date.setHours(0, dateTimeMinutes, 0, 0)
    console.log(date)
}
```

Please close the issue when solved!
